src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EmbeddedObjectNode.cs

```diff
@@ -30,7 +30,6 @@ public virtual int Offset
             }
             set
             {
-                Debug.Assert(_offset == InvalidOffset || _offset == value);
```

Comment:
Why was this assert removed? It means that there is something pretty wrong going on.